src/engine/Service.cpp

```diff
@@ -200,6 +171,58 @@ ResultTable Service::computeResult() {
   return {std::move(idTable), resultSortedOn(), std::move(localVocab)};
 }
 
+std::optional<std::string> Service::getSiblingValuesClause() const {
+  if (siblingTree_ == nullptr) {
+    return std::nullopt;
+  }
+
+  const auto& siblingResult = siblingTree_->getResult();
+  if (siblingResult->size() >
+      RuntimeParameters().get<"service-max-value-rows">()) {
+    return std::nullopt;
+  }
+
+  std::vector<ColumnIndex> commonColumnIndices;
+  const auto& siblingVars = siblingTree_->getVariableColumns();
+  std::string vars = "";
```

Comment:
There is one bug remaining:
If a row (combination of all relevant variables) appears multiple times in the sibling result, then you may only store it once in the VALUES clause, otherwise the SERVICE will duplicate each result leading to wrong results with too many rows.